cmd/relay/p2p.go

```diff
@@ -16,37 +16,46 @@ import (
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/libp2p/go-libp2p/p2p/protocol/circuitv2/relay"
 	ma "github.com/multiformats/go-multiaddr"
+	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/obolnetwork/charon/app/errors"
 	"github.com/obolnetwork/charon/app/log"
 	"github.com/obolnetwork/charon/app/peerinfo"
+	"github.com/obolnetwork/charon/app/promauto"
 	"github.com/obolnetwork/charon/app/z"
 	"github.com/obolnetwork/charon/p2p"
 )
 
 // startP2P returns a started libp2p host or an error.
-func startP2P(ctx context.Context, config Config, key *k1.PrivateKey, reporter metrics.Reporter) (host.Host, error) {
+func startP2P(ctx context.Context, config Config, key *k1.PrivateKey, reporter metrics.Reporter) (host.Host, *prometheus.Registry, error) {
 	if len(config.P2PConfig.TCPAddrs) == 0 {
-		return nil, errors.New("p2p TCP addresses required")
+		return nil, nil, errors.New("p2p TCP addresses required")
 	}
 
 	if config.LibP2PLogLevel != "" {
 		if err := libp2plog.SetLogLevel("relay", config.LibP2PLogLevel); err != nil {
-			return nil, errors.Wrap(err, "set relay log level")
+			return nil, nil, errors.Wrap(err, "set relay log level")
 		}
 		if err := libp2plog.SetLogLevel("rcmgr", config.LibP2PLogLevel); err != nil {
-			return nil, errors.Wrap(err, "set rcmgr log level")
+			return nil, nil, errors.Wrap(err, "set rcmgr log level")
 		}
 	}
 
 	tcpNode, err := p2p.NewTCPNode(ctx, config.P2PConfig, key, p2p.NewOpenGater(), config.FilterPrivAddrs,
 		libp2p.ResourceManager(new(network.NullResourceManager)), libp2p.BandwidthReporter(reporter))
 	if err != nil {
-		return nil, errors.Wrap(err, "new tcp node")
+		return nil, nil, errors.Wrap(err, "new tcp node")
 	}
 
 	p2p.RegisterConnectionLogger(ctx, tcpNode, nil)
 
+	labels := map[string]string{"relay_peer": p2p.PeerName(tcpNode.ID())}
+	log.SetLokiLabels(labels)
+	promRegistry, err := promauto.NewRegistry(labels)
```

Comment:
Isn't it a bit weird creating a prometheus registry in `startP2P` function? Does it make sense to create it outside of it (in `cmd/relay/relay.go:Run` afaict) and pass it as an argument, similarly to how the reporter is passed (another monitoring structure)?

Reply:
The problem here is in the dependency on `labels` which we can populate only after we started a tcpNode.
I would rework it in a different way to remove this dependency, but would be a drastic change, not for the scope of this PR for sure.